XavierAubert/core-v-verif
# Add new interrupt and debug tests

This PR is for interrupt and debug tests:

1. Add new exception test stream for hwloop and corresponding test and regress file updates
2. Add new test_cfg for running debug and interrupt tests
3. Update exsting corev-dv pulp test's corev-dv.yaml to resolve issues due to enable_floating_point or ebreak configs
4. Update xpulp_rand_test regress yaml to include all new tests
5. Add new functions push_regfile_to_kernel_stack and pop_regfile_from_kernel_stack to include fpr along with gpr.
6. Update existing push{pop}_fpr*_debugger_stack functions to check for mstatus.FS
7. Override all handler functions to use new push-pop functions

New Commits - Aug 31st

8. Fix issue with DP (debug pointer, used for debug rom) getting corrupted for pulp random tests - Fix my adding constraints for available regs and using reserved_regs from config object to apply reg randomization constraints instead of local list. Update avaliable reg count for rand by 1 for dp reg.
9. Updates for cfg object - set/get cv32e40p_cfg (extended cfg) object from test and sequence using uvm_config_db. Randomizing this config by overriding the base test randomize_cfg()
10. Improve debug single step test for increased number on step iterations
11. Include all instructions (including xpulp) in debug rom random instructions
